src/OrchardCore/OrchardCore.ContentManagement.GraphQL/Queries/Types/DynamicContentTypeBuilder.cs

```diff
@@ -84,45 +90,66 @@ public void Build(ISchema schema, FieldType contentQuery, ContentTypeDefinition
                 else
                 {
                     // Check if another builder has already added a field for this part.
-                    var existingField = contentItemType.GetField(partName.ToFieldName());
-                    if (existingField != null)
+                    var partFieldName = partName.ToFieldName();
+                    var partFieldType = contentItemType.GetField(partFieldName);
+
+                    if (partFieldType != null)
                     {
-                        // Add content field types.
-                        foreach (var field in part.PartDefinition.Fields)
+                        // Add dynamic content field types to the static part type.
+                        var partContentItemType = schema.AdditionalTypeInstances
+                            .OfType<IObjectGraphType>()
+                            .Where(type => type.GetType() == partFieldType.Type)
+                            .FirstOrDefault();
+
+                        if (partContentItemType != null)
                         {
-                            foreach (var fieldProvider in contentFieldProviders)
+                            foreach (var field in part.PartDefinition.Fields)
                             {
-                                var contentFieldType = fieldProvider.GetField(schema, field, part.Name);
-
-                                if (contentFieldType != null && !contentItemType.HasField(contentFieldType.Name))
+                                foreach (var fieldProvider in contentFieldProviders)
                                 {
-                                    contentItemType.AddField(contentFieldType);
-                                    break;
+                                    var contentFieldType = fieldProvider.GetField(schema, field, part.Name);
+
+                                    if (contentFieldType != null)
+                                    {
+                                        if (_contentOptions.ShouldSkip(contentFieldType.Type, contentFieldType.Name))
+                                        {
+                                            continue;
+                                        }
+
+                                        if (partContentItemType.HasFieldIgnoreCase(contentFieldType.Name))
+                                        {
+                                            continue;
+                                        }
```

Comment:
```suggestion
                                        if (_contentOptions.ShouldSkip(contentFieldType.Type, contentFieldType.Name) || partContentItemType.HasFieldIgnoreCase(contentFieldType.Name))
                                        {
                                            continue;
                                        }

```

Reply:
Anything prevent us to combine both conditions into one?

Reply:
No, I kept the separation for clarity only.

Reply:
I suggest combining them why the conditions are not too big and quite clear